users/views.py

```diff
@@ -44,8 +45,9 @@ def profile(request: 'HttpRequest') -> HttpResponse:
     """
     try:
         uid = int(request.GET.get('id', request.user.id))
-        prof = UserProfile.objects.get_or_create(user_id=uid)[0]
-    except (ValueError, IntegrityError) as e:
+        user = User.objects.get(pk=uid)
```

Comment:
It doesn't though, actually if you just set the `user_id` manually it will create the profile even if the user doesn't exist! So the code will succeed successfully even if you try to view the `UserProfile` of a `User` that doesn't exist. In my test case I use id of 5 for example. But the reference to the `User` model will be completely broken, so in the next if clause, when `prof.user.is_superuser` gets called, THEN `User.DoesNotExist` will get thrown, and the view will break.

So it is not only necessary for the view to function but also for the underlying intended behaviour as well.

Reply:
If you wish to investigate further, feel free to revert the change locally and look at the `TestProfile.test_invalid_user` test case where the issue caused by not fixing this is covered.

Reply:
That… doesn't make sense.

Reply:
Potential fix: https://github.com/pytest-dev/pytest-django/issues/754#issuecomment-519081241